app/controllers/comments_controller.rb

```diff
@@ -10,10 +10,13 @@ def create
 
     if @comment.valid?
       @comment.save
-      redirect_to user_post_path(id: @post.id, user_id: @user.id)
+      # redirect_to user_post_path(id: @post.id, user_id: @user.id)
+      flash.notice = 'Comment added successfuly.'
     else
-      redirect_to new_user_post_comment_path(@user.id, @post.id)
+      # redirect_to new_user_post_comment_path(@user.id, @post.id)
+      flash.alert = 'Please try again.'
     end
+    redirect_to new_user_post_comment_path(@user.id, @post.id)
```

Comment:
- [ ]  Kindly redirect to the comment#new form only if the comment failed to be created because is where your flash messages should be rendered. hint :point_down: 

```rb
  def create
    @user = User.find(params[:user_id])
    @post = Post.find(params[:post_id])
    @comment = Comment.create(post: @post, author: @user, text: params[:comment][:text])

    if @comment.valid?
      @comment.save
      flash.notice = 'Comment added successfuly.'
      redirect_to user_post_path(@user.id, @post.id)
    else
      flash.alert = 'Please try again.'
      redirect_to new_user_post_comment_path(@user.id, @post.id)
    end
  end
```